.github/workflows/test.yml

```diff
@@ -108,3 +108,10 @@ jobs:
               -exec echo '==== {} ====' \; -exec cat '{}' \;
           find "${TMPDIR:-/tmp}/${USER}/cylctb-"* -type f \
               -exec echo '==== {} ====' \; -exec cat '{}' \;
+
+      - name: Upload
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: cylc-run
+          path: cylc-run
```

Comment:
Added this to help with debugging flaky tests. It should upload the cylc-run directory as an [artefact](https://docs.github.com/en/free-pro-team@latest/actions/guides/storing-workflow-data-as-artifacts) that we can download and inspect locally. Artefacts are housekept along with build logs after 90 days.

Reply:
Nice, good idea.

Reply:
If you give the `name: Test` step an `id: `, we can check the failure of that step specifically instead of the installation steps. Not really necessary though as most of the time it's the test step that fails.